pkg/virt-controller/watch/vmi_test.go

```diff
@@ -1769,6 +1769,17 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			addVirtualMachine(vmi)
 
+			if phase == virtv1.Running {
```

Comment:
You want to replace this by 
```
   pod := <some minimalistic Pod spec>
   kubeClient ... Create(pod)
```
and then you don't need a reactor. Think about kubeClient as an API server and you want to populate it with the `virt-handler` Pod.

Reply:
IIUC, the idea is to create a `virt-handler` pod in this test so we can avoid the reactor. However, this is not possible due to the fact that in this test the `virt-launcher` is not present, and it will call the new `isVirtHandlerAlive` function, which needs to list the virt-handler pods. Therefore, the reactor is still needed.
Another possibility is to create `virt-launcher` to avoid the reactor (the `virt-launcher` will be present and the `isVirtHandlerAlive` won't be called). However, I'm unsure if the test scenario wants to have the `virt-launcher` pod. WDYT?